---
title: "Exploratory Analysis of Breast Cancer Data"
author: "Alan Selewa"
date: 2017-11-17
output: html_document
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=6, fig.path='Figs/',
                      echo=TRUE, warning=FALSE, message=FALSE, comment = NA)
```

```{r include=FALSE}
library(kableExtra)
```

## Load data

Single-cell RNA-seq across 33 cells were aligned to ``hg19`` using ``STAR`` on the RCC and 33 BAM files were produced as a result. Each cell has about $10^5$ unique reads. The average coverage is **16.69**, with **9.7%** of transcriptome covered at atleast **10X**. Next, I used ``bam-readcounts`` (default parameters) to obtain read counts at each mutation positions for each nucleotide. The following data structure was used to summarise single cell mutation profiles:

```{r}
source('/Users/aselewa/sc_call_variants/readcounts_parser.R')
knitr::kable(covmat,"html") %>% kable_styling() %>% scroll_box(width="805px",height='400px')
```

The frequency of mutation types is shown below. We have 170 coding mutations, not inlucding the UTR regions.

```{r}
table(covmat$Type)
```


We now look at the mutational profiles of some genes across all cells.

## Basic exploration

Using the single-cell digital expression matrix, the top 20 expressed mutated genes in these cells are:

```{r}
geneCounts <- rowSums(tpms[covmat[,"Gene"],])
geneCounts[order(geneCounts,decreasing=TRUE)][1:20]
```

We can visualize the reference and alternative counts for each of these genes across all cells. We show only the genes with exon mutations.

```{r}
pos <- seq(8,length(covmat),by=4)
freqs <- cbind(avgRef = unname(unlist(covmat[covmat$Gene=='EIF5',pos])), 
               avgAlt = unname(unlist(covmat[covmat$Gene=='EIF5',pos+1])),
               TPM = 0.1*(unname(unlist(covmat[covmat$Gene=='EIF5',pos-2]))))
barplot(t(freqs),beside=TRUE,legend.text = c("Ref","Alt","TPTM"),
        col = c("red","blue","black"),xlab='Cell',ylab='Counts',main='EIF5')
```

```{r}
freqs <- cbind(avgRef = unname(unlist(covmat[covmat$Gene=='APLP2',pos])), 
               avgAlt = unname(unlist(covmat[covmat$Gene=='APLP2',pos+1])))
barplot(t(freqs),beside=TRUE,legend.text = c("Ref","Alt"),
        col = c("red","blue"),xlab='Cell',ylab='Counts',main='APLP2')
```


```{r}
freqs <- cbind(avgRef = unname(unlist(covmat[covmat$Gene=='ILK',pos])), 
               avgAlt = unname(unlist(covmat[covmat$Gene=='ILK',pos+1])))
barplot(t(freqs),beside=TRUE,legend.text = c("Ref","Alt"),
        col = c("red","blue"),xlab='Cell',ylab='Counts',main='HSPA4')
```

Even for the most expressed genes, the counts at the mutated positions are not consistent across cells. 

## Error Rates

$$ Err = \frac{\sum_i^{N} A_i}{\sum_i^{N} (R_i+A_i)} $$

```{r}
errors <- gen.MultiErrorRate()
hist(errors,main='Distribution of errors (N=21)')
```

The mean error rate is
```{r}
mean(errors)
```
