---
title: "Phylogenetic Analysis of BC Data"
author: "Alan Selewa"
date: 2017-11-17
output: html_document
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=10, fig.height=6, fig.path='Figs/',
                      echo=TRUE, warning=FALSE, message=FALSE, comment = NA)
```

## Load processed data

We load the familiar data structure for single-cell mutation profiles.

```{r}
source("/Users/aselewa/sc_call_variants/readcounts_parser.R")
covmat <- gen.CovMat()
```

## Mutation Matrix

For each cell-mutation, we count the total number of reference and alternative alleles. Under the null model, the probability of observing n alternative alleles and N-n reference alleles is given by:

$$ P(A = n) = {{N}\choose{n}}  p_{error}^{n} (1 - p_{error})^{(N-n)} $$

where $p_{error} = 0.004$ is the estimated error rate. The error rate is define as the probability of observing a mutation by chance. We perform a binomial test to obtain the p-value of our observed alternative/reference counts under the null model. We choose a cut-off of 0.05. If a cell-mutation satisfies the cutoff, then a "1" is placed in the mutation matrix and 0 otherwise. Mutations with missing data are labeled as 3. 

```{r}
phylomat <- gen.PhyMat(covmat)
```

Using top 30 mutations (in terms of least missing data) on 33 cells, 66% of the entries in the mutation matrix are missing values. This is equivalent to effectively halfing the number of samples (cells) when using SCITE to reconstruct the mutation matrix. Using SCITE, we obtain 9 co-likely trees. We visualize one of them below:

```{r}
library(DiagrammeR)
grViz('/Users/aselewa/tree_inference/old/bc03_tree_ml0.gv')
```

## CNV Profiles

It is difficult to distinguish copy-number variation from differential expression in scRNA-seq data. Instead, researchers thus far detect large-scale CNVs by ordering genes by their genomic position and performing a moving average over ~150 gene blocks, depending on the number of genes used in the analysis.

This idea was executed in [Patel et al.](http://science.sciencemag.org/content/early/2014/06/11/science.1254257) and [Chung et al.](https://www.nature.com/articles/ncomms15081) using single-cell RNA-seq data. We adopt this idea to try to infer large-scale CNVs.

As a reference, we use GTEx normal breast tissue TPMs. We average across all breast tissue. A CNV reference profile is constructed in parallel using 150 gene blocks.

The CNV profile for each cell is substracted from the reference CNV and the result is shown below. The black vertical bars seperate chromosomes. The order of chromosomes is from left to right as 1, 2, 3, .. ,X, Y. I used all the genes that occur in both GTEx and the single-cell data (~10k genes), with a gene block of 150 genes. The dendrogam on the left represents heirarichal clustering of the expression profiles. 

```{r bc03_cnv_plot}
source('/Users/aselewa/sc_call_variants/inferCNV.R')
results <- 
```


These CNV profiles are consistent with expected amplification/deletion of chromosomes in ER+ breast cancer. In particular, chromosomes 8 and 17 amplification is predictive of breast cancer. Another observation is that we have two distant clusters here. It seems to suggest that half the cells are normal cells because they are closer to the reference CNV from GTEx.


We have complete phylogeny for 4 cells in the mutation matrix, namely, BC03_13, BC03_28, BC03_50, and BC03_58. According to their mutation profiles, 13 is distant from 28,50,and 58, while these 3 cells have 0 distance between them. Does this pattern in the heirarchical clustering of CNV profiles hold? Yes, the trend holds. 

![](/Users/aselewa/web/img/cnv_mutation_fig.png)

The figure above compares the two trees. The left tree is a little finer due to the nature of the data, but overall the trend holds. Perhaps we can use this CNV data to infer missing data in the original phylogenetic reconstruction problem.

For instance, BC03_57 has the following mutational profile:

```{r}
subphylo["BC03_57",]
```

From the CNV profiles, BC03_57 shares a more recent ancestor with BC03_50, BC03_28, and BC058, thus the missing data is likely to be '1', because the CNV tree tells us it is further from BC03_13 than BC03_58/28/50.

