<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Alan Selewa" />

<meta name="date" content="2017-11-17" />

<title>Phylogenetic Inference with SNVs and CNVs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Alan Selewa</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="cv_page.html">
    <span class="fa fa-file  fa-2x"></span>
     
    CV
  </a>
</li>
<li>
  <a href="research.html">
    <span class="fa fa-book fa-2x"></span>
     
    Research
  </a>
</li>
<li>
  <a href="gallery.html">
    <span class="fa fa-camera fa-2x"></span>
     
    Gallery
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/aselewa">
    <span class="fa fa-github fa-2x"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/alan-selewa-000889b8/">
    <span class="fa fa-linkedin fa-2x"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Phylogenetic Inference with SNVs and CNVs</h1>
<h4 class="author"><em>Alan Selewa</em></h4>
<h4 class="date"><em>2017-11-17</em></h4>

</div>


<div id="sample-1-bc03" class="section level2">
<h2>Sample 1 (BC03)</h2>
<div id="clustering-cells" class="section level3">
<h3>Clustering Cells</h3>
<p>Using a window of 150 genes, the relative CNV profiles of single cells are computed and clustered.</p>
<pre class="r"><code>source(&#39;inferCNV.R&#39;)
results03 = clusterCNVs(&#39;03&#39;)</code></pre>
<p><img src="Figs/cluster_cnvs_bc03-1.png" width="960" /></p>
<pre class="r"><code>bc03_cnv = results03[[1]]
cells03 = results03[[2]]
genes03 = results03[[3]]</code></pre>
<p>Next, we plot first and second principle component of the expression matrix.</p>
<p><img src="Figs/bc03_pca-1.png" width="960" /></p>
<p>Based on the CNV plot above, chromosome 8 and 17 are both amplified in the top cluster. Cells within this cluster also seperate in the PC analysis. Thus, these are the cancerous cells, while the other cells are healthy cells.</p>
<p>Using the tree generated by the CNV profiles above, we look at how well oncogenes/tumor suppressor genes agree with the ordering obtained by CNV clustering. The oncogenes ERBB2/ERBB3 are on chromosome 6, and MYC which is on chromosome 8. Both of these chromosomes are amplified for unhealthy cells. The heatmap below confirms that the expression of these oncogenes is also elevated.</p>
<p><img src="Figs/bc03_cnv_ggtree-1.png" width="960" /></p>
</div>
<div id="mutation-matrix" class="section level3">
<h3>Mutation matrix</h3>
<p>We now turn to trees generated by SNVs for the same set of data. Given what we know about these cells, we can benchmark the results of the SNV analysis. In particular, we know that these cells are a mix of unhealthy and healthy, so at the very least, we expect the SNV tree to seperate these two populations, as the cancer cells contain mutations that the healthy ones do not.</p>
<p>We load the familiar data structure for single-cell mutation profiles.</p>
<pre class="r"><code>source(&quot;readcounts_parser.R&quot;)
covmat &lt;- gen.covmat(&#39;03&#39;)</code></pre>
<p>As a reminder: for each cell-mutation, we count the total number of reference and alternative alleles. Under the null model, the probability of observing n alternative alleles and N-n reference alleles is given by:</p>
<p><span class="math display">\[ P(A = n) = {{N}\choose{n}}  p_{error}^{n} (1 - p_{error})^{(N-n)} \]</span></p>
<p>where <span class="math inline">\(p_{error} = 0.004\)</span> is the estimated error rate. The error rate is define as the probability of observing a mutation by chance. We perform a binomial test to obtain the p-value of our observed alternative/reference counts under the null model. We choose a cut-off of 0.05. If a cell-mutation satisfies the cutoff, then a “1” is placed in the mutation matrix and 0 otherwise. Mutations with missing data are labeled as 3.</p>
<pre class="r"><code>phylomat &lt;- gen.mutmat(covmat)
head(phylomat[1:10,1:5])</code></pre>
<pre><code>        BC03_03 BC03_06 BC03_09 BC03_11 BC03_13
EIF5          3       3       0       1       1
HUWE1         3       3       1       1       0
CAGE1         3       3       1       3       0
Unknown       1       1       3       1       1
TPR           3       0       0       0       1
HSPA6         3       3       0       3       3</code></pre>
<p>Using top 27 mutations (in terms of least missing data) on 33 cells, 66% of the entries in the mutation matrix are missing values. This is equivalent to effectively halfing the number of samples (cells) when using SCITE to reconstruct the mutation matrix.</p>
</div>
<div id="scite-output" class="section level3">
<h3>SCITE Output</h3>
<p>The result from SCITE is shown below, where all the branch lengths have been set equal. The heatmap next to the tree shows 5 oncogenes and 4 tumor suppressor genes and its expression for each cell.</p>
<p><img src="Figs/scite_output_bc03-1.png" width="960" /></p>
</div>
</div>
<div id="sample-2-bc08" class="section level2">
<h2>Sample 2 (BC08)</h2>
<p>The sample above is not so interesting because there seems to be only one clone and a healthy subset. It would be interesting to run this analysis on data where multiple clones exist.</p>
<div id="clustering-cells-1" class="section level3">
<h3>Clustering Cells</h3>
<p>Based on the CNV profiles below, patient BC03 may have multiple clones.</p>
<pre class="r"><code>source(&#39;inferCNV.R&#39;)
result02 = clusterCNVs(&#39;08&#39;)</code></pre>
<p><img src="Figs/bc08_cnv-1.png" width="960" /></p>
<pre class="r"><code>bc02_cnv = result02[[1]]
cells02 = result02[[2]]
genes02 = result02[[3]]</code></pre>
<p><img src="Figs/bc08_pca-1.png" width="960" /></p>
<p><img src="Figs/bc08_cnv_ggtree-1.png" width="960" /></p>
</div>
<div id="scite-output-1" class="section level3">
<h3>SCITE Output</h3>
<p>In progress.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
