<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Alan Selewa</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="cv_page.html">
    <span class="fa fa-file  fa-2x"></span>
     
    CV
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book fa-2x"></span>
     
    Research
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="research.html">Overview</a>
    </li>
    <li>
      <a href="dndq.html">Single Cell Analysis</a>
    </li>
    <li>
      <a href="protein_clusters.html">Protein Evolution</a>
    </li>
  </ul>
</li>
<li>
  <a href="gallery.html">
    <span class="fa fa-camera fa-2x"></span>
     
    Gallery
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/aselewa">
    <span class="fa fa-github fa-2x"></span>
     
  </a>
</li>
<li>
  <a href="https://www.linkedin.com/in/alan-selewa-000889b8/">
    <span class="fa fa-linkedin fa-2x"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<hr />
<div id="protein-clusters" class="section level1">
<h1><strong>Protein Clusters</strong></h1>
<hr />
<div id="modeling-protein-cluster-diversity-in-pangenomic-data" class="section level2">
<h2>Modeling Protein Cluster Diversity in Pangenomic Data</h2>
<p>Given a multi-sequence alignment, or a protein block, we wish to compute the diversity of the protein block. One motivation for doing this is to flag a group of homologues proteins that are undergoing a sweep. Suppose there are <span class="math inline">\(N\)</span> aligned sequences in the protein block, then the vector of residues at site <span class="math inline">\(i\)</span> is represented by <span class="math inline">\(X_i\)</span>. We assume these <span class="math inline">\(N\)</span> sequences started from a common ancestor, and that they are equally far away from the ancestor in time. All the sequences have length <span class="math inline">\(m\)</span> (for now).</p>
<p>In order to quantify diversity, we first define a model. We use a first-order Markov model to calculate the likelihood of a vector <span class="math inline">\(X_i\)</span> having mutated from an ancestor <span class="math inline">\(X^*\)</span>, and we margenalize over the ancestors using all 20 amino acids. The likelihood of a given vector of residues <span class="math inline">\(X_i\)</span>, paramterized by the Markov chain <span class="math inline">\(\theta\)</span> is given by:</p>
<p><span class="math display">\[L(X_i|\theta) = \sum_{X^*} P(X^*) \prod_{j=1}^N P(X_{i,j}|X^*)\]</span> where <span class="math inline">\(P(X^*)\)</span> is the probability of that the ancestor is <span class="math inline">\(X^*\)</span>, which is obtained from the stationary distribution of the Markov chain <span class="math inline">\(\theta\)</span>. The conditionary distribution <span class="math inline">\(P(X_{i,j}|X^*)\)</span> is simply given by the Markov chain <span class="math inline">\(\theta\)</span>.</p>
<center>
<div class="figure">
<img src="img/mc.png" />

</div>
</center>
<p>Now we can perform likelihood ratio testing. In particular, we can define a score:</p>
<p><span class="math display">\[S = \log{\frac{L(X_i|\theta&#39;)}{L(X_i|\theta)}}\]</span> where <span class="math inline">\(\theta&#39;\)</span> and <span class="math inline">\(\theta\)</span> are the alternative hypothesis and null hypothesis, respectively. For <span class="math inline">\(\theta\)</span>, we can choose codon substitution matrices if we wish to test evolution or empirical matrices such as BloSum to test how conserved a region is. However, the alternative hypotehsis <span class="math inline">\(\theta&#39;\)</span> will be quite sparse if calculated from the given protein blocks as there is not a lot of data.</p>
<p>Instead, we take an alternative strategy where we alter the hypothesis <span class="math inline">\(\theta\)</span> in a way that increases the likelihood. This technique is typically referred to Phylogeny by Maximum Likelihood (PAML).</p>
<p>We begin with a BloSum90 conditional distribution matrix, which henceforth I refer to as a transition matrix (<span class="math inline">\(\theta\)</span>). Using this transition matrix, we compute the instantaneous rate matrix using the relationship <span class="math inline">\(\theta(t) = \exp(Rt)\)</span>:</p>
<p><span class="math display">\[R = \frac{1}{t^*} \log{\theta} = \frac{1}{t^*} T^{-1} \log{(\Lambda}) T\]</span> where <span class="math inline">\(T^{-1}\Lambda T\)</span> is the eigendecomposition of <span class="math inline">\(theta\)</span>. Next, we modify <span class="math inline">\(R\)</span> by multiplying it by a non-synonymous transition rate <span class="math inline">\(\omega\)</span>. Using the modified instantaneous rate matrix, we re-compute <span class="math inline">\(\theta\)</span>:</p>
<p><span class="math display">\[\theta(\omega) = \exp(R\omega) = D^{-1} \exp(M) D\]</span> where we have used eigendecomposition again to compiute the exponential of matrix <span class="math inline">\(R\omega\)</span>. The value <span class="math inline">\(t*\)</span> represents the divergence time between the sequences used to compute BloSum90, however, we do not know what this value is, thus we set <span class="math inline">\(t^* = 1\)</span> and use <span class="math inline">\(\omega\)</span> in units of <span class="math inline">\(t^*\)</span>. Now that we have <span class="math inline">\(\theta(\omega)\)</span>, we can compute the likelihood and find the non-synonymous transition rate that maximizes the likelihood of a residue vector <span class="math inline">\(X_i\)</span>, which we call <span class="math inline">\(\hat{\omega}\)</span>:</p>
<p><span class="math display">\[\hat{\omega_i} = \underset{\omega}{\arg\max} \{ L(X_i|\theta(\omega))\}\]</span> Intuitively, if <span class="math inline">\(\hat{\omega} &gt; 1\)</span> for a certain <span class="math inline">\(X_i\)</span>, then <span class="math inline">\(X_i\)</span> is less likely to occur than predicted by BloSum. Similarly, <span class="math inline">\(\hat{\omega} &lt; 1\)</span> indicates that <span class="math inline">\(X_i\)</span> is more likely to occur than predicted by BloSum. Values of <span class="math inline">\(\hat{\omega} = 1\)</span> are expected under BloSum.</p>
<p><span class="math display">\[ S_i = \log{\frac{L(X_i|\omega = \hat{\omega})}{L(X_i|\omega = 1)}} \]</span> and the total diversity would be:</p>
<p><span class="math display">\[ D = \sum_{i=1}^m S_i\]</span></p>
<div id="short-example" class="section level3">
<h3>Short example</h3>
<p>Suppose there are 3 protein sequences, and the <span class="math inline">\(i\)</span>th position is polymorphic:</p>
<p><span class="math display">\[X_i = \begin{pmatrix} A \\ A \\ W \end{pmatrix}\]</span></p>
<p>The likelihood that these 3 sequences came from a single ancestor <span class="math inline">\(X^*\)</span> is: ,</p>
<p><span class="math display">\[ L(X_i|\theta) = P(A) P(A|A)^2P(A|W) + P(R)P(A|R)^2P(A|W) + \ldots + P(V)P(A|V)^2P(W|V) \]</span> Some observations:</p>
<ol style="list-style-type: decimal">
<li><p>As <span class="math inline">\(\omega\)</span> increases, non-synonymous transitions such as <span class="math inline">\(P(A|W)\)</span> will increases, whereas terms such as <span class="math inline">\(P(A|A)\)</span> decrease due to the normalization constraint, i.e. <span class="math inline">\(\sum_{i=1}^{20} P(A|X_i) = 1\)</span>. Thus, <span class="math inline">\(\omega\)</span> cannot grow arbitrarily large.</p></li>
<li><p>Synonymous transitions will always have <span class="math inline">\(\hat{\omega} = 1\)</span>, and thus <span class="math inline">\(S_i = 0\)</span>.</p></li>
</ol>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
